### ISSUE TITLE
How to render `mdx` to `html` from filesystem?

### ISSUE THREAD

**Issue by geocine** [zbysir/gojsx#1]
I tried this but it doesn't work. I think it expects the file to be served. I want to render the mdx before serving so I want to load from filesystem.

```go
func renderMDX(path string) (string, error) {

	// initialize error variable
	var err error

	j, jsxError := gojsx.NewJsx(gojsx.Option{})
	if jsxError != nil {
		err = jsxError
	}
	// This path is filesystem path like `C:/somefolder/index.mdx
	s, renderError := j.Render(path, nil)
	if renderError != nil {
		err = renderError
	}

	return s, err
}
```

**Comment by zbysir** [comment]
What is the error message you get? @geocine

**Comment by zbysir** [comment]
[Here](https://github.com/zbysir/gojsx/blob/master/jsx_test.go#L184) is an example that renders mdx and it works.

Tip: mdx is super complex, what I have implemented is just a subset, you can start experimenting, if there are new features that need support then I am willing to implement them.

**Comment by geocine** [comment]
Right apologies I did not post any error details. I will isolate the code and check. I will get back to you.

**Comment by geocine** [comment]
This is my mdx file https://gist.github.com/geocine/c71af743c0085071da28505fcd50a14e

This is how I run the code

```golang
// gojsx is https://github.com/zbysir/gojsx

package main

import (
	"fmt"

	"github.com/zbysir/gojsx"
)

func renderMDX(path string) (string, error) {

	// initialize error variable
	var err error

	j, jsxError := gojsx.NewJsx(gojsx.Option{})
	if jsxError != nil {
		err = jsxError
	}
	// This path is filesystem path like `C:/somefolder/index.mdx
	s, renderError := j.Render(path, nil)
	if renderError != nil {
		err = renderError
	}

	return s, err
}

// main

func main() {

	// path to mdx file
	path := "./test.mdx"

	// render mdx file
	s, err := renderMDX(path)
	if err != nil {
		panic(err)
	}

	// print rendered html
	fmt.Println(s)
}

```

This is the error output

```sh
PS D:\PL\gojsx-test> go run main.go
panic: display: flex; justify-content: center;

goroutine 1 [running]:
github.com/zbysir/gojsx.renderStyle(0xffffffffff?, {0x122ae60?, 0xc00007f860?})
        C:/Users/Aivan/go/pkg/mod/github.com/zbysir/gojsx@v0.4.7/jsx.go:801 +0x91
github.com/zbysir/gojsx.renderAttributes.func1({0xc0002ebd6a, 0x5}, {0x122ae60, 0xc00007f860})
        C:/Users/Aivan/go/pkg/mod/github.com/zbysir/gojsx@v0.4.7/jsx.go:712 +0x1a5
github.com/zbysir/gojsx.sortMap(0xc000164e10, 0xc00065fa28)
        C:/Users/Aivan/go/pkg/mod/github.com/zbysir/gojsx@v0.4.7/jsx.go:676 +0x203
github.com/zbysir/gojsx.renderAttributes(0xc00065fdc0, 0xc000164e10)
        C:/Users/Aivan/go/pkg/mod/github.com/zbysir/gojsx@v0.4.7/jsx.go:698 +0x1f0
github.com/zbysir/gojsx.render(0xc00065fdc0, {0x1255880?, 0xc000164d80?})
        C:/Users/Aivan/go/pkg/mod/github.com/zbysir/gojsx@v0.4.7/jsx.go:948 +0x7f2
github.com/zbysir/gojsx.render(0xc00065fdc0, {0x1222a40?, 0xc000144a68?})
        C:/Users/Aivan/go/pkg/mod/github.com/zbysir/gojsx@v0.4.7/jsx.go:898 +0xf35
github.com/zbysir/gojsx.render(0xc00065fdc0, {0x12b7160?, 0xc000164510?})
        C:/Users/Aivan/go/pkg/mod/github.com/zbysir/gojsx@v0.4.7/jsx.go:929 +0x31d
github.com/zbysir/gojsx.Render(...)
        C:/Users/Aivan/go/pkg/mod/github.com/zbysir/gojsx@v0.4.7/jsx.go:884
github.com/zbysir/gojsx.VDom.Render(0x1279d60?)
        C:/Users/Aivan/go/pkg/mod/github.com/zbysir/gojsx@v0.4.7/jsx.go:879 +0x34
github.com/zbysir/gojsx.(*Jsx).Render(0x1445510?, {0x12ff3c6, 0xa}, {0x0, 0x0}, {0x0, 0x0, 0x1a235d0?})
        C:/Users/Aivan/go/pkg/mod/github.com/zbysir/gojsx@v0.4.7/jsx.go:319 +0x313
main.renderMDX({0x12ff3c6, 0xa})
        D:/PL/gojsx-test/main.go:21 +0x65
main.main()
        D:/PL/gojsx-test/main.go:37 +0x29
exit status 2
```

**Comment by zbysir** [comment]
@geocine It is fixed in v0.4.8, please update gojsx by following command

```
go get github.com/zbysir/gojsx@v0.4.8
```

**Comment by geocine** [comment]
It works, that was fast. Thank you!
